Description: Include more XDG menu categories when parsing with wmmenugen.
Origin: http://repo.or.cz/wmaker-crm.git/commit/c261f8e
Origin: http://repo.or.cz/wmaker-crm.git/commit/ac0690b
Origin: http://repo.or.cz/wmaker-crm.git/commit/adc1687
Origin: http://repo.or.cz/wmaker-crm.git/commit/ff865c3
Last-Update: 2017-07-09

--- a/util/wmmenugen_parse_xdg.c
+++ b/util/wmmenugen_parse_xdg.c
@@ -178,6 +178,9 @@
 			getMenuHierarchyFor(&xdg->Category);
 		}
 
+		if (xdg->Category == NULL)
+			xdg->Category = wstrdup(_("Other"));
+
 		wfree(key);
 		key = NULL;
 	}
@@ -523,6 +526,9 @@
 		} else if (strcmp(p, "Office") == 0) {
 			snprintf(buf, sizeof(buf), "%s", _("Office"));
 			break;
+		} else if (strcmp(p, "Science") == 0) {
+			snprintf(buf, sizeof(buf), "%s", _("Science"));
+			break;
 		} else if (strcmp(p, "Settings") == 0) {
 			snprintf(buf, sizeof(buf), "%s", _("Settings"));
 			break;
@@ -532,13 +538,26 @@
 		} else if (strcmp(p, "Utility") == 0) {
 			snprintf(buf, sizeof(buf), "%s", _("Utility"));
 			break;
+		/* reserved categories */
+		} else if (strcmp(p, "Screensaver") == 0) {
+			snprintf(buf, sizeof(buf), "%s", _("Screensaver"));
+			break;
+		} else if (strcmp(p, "TrayIcon") == 0) {
+			snprintf(buf, sizeof(buf), "%s", _("Tray Icon"));
+			break;
+		} else if (strcmp(p, "Applet") == 0) {
+			snprintf(buf, sizeof(buf), "%s", _("Applet"));
+			break;
+		} else if (strcmp(p, "Shell") == 0) {
+			snprintf(buf, sizeof(buf), "%s", _("Shell"));
+			break;
 		}
 		p = strtok(NULL, ";");
 	}
 
 
 	if (!*buf)		/* come up with something if nothing found */
-		snprintf(buf, sizeof(buf), "%s", _("Applications"));
+		snprintf(buf, sizeof(buf), "%s", _("Other"));
 
 	*xdgmenuspec = wstrdup(buf);
 }
